<a href="https://colab.research.google.com/github/beautyoctavia1210-dot/FlashSale/blob/main/EDA_KKA.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

#**Latihan 1**

In [ ]:
import numpy as np

harga = np.array([5000, 7000, 3000, 12000, 4500])
print('Rata-rata harga: ', harga.mean())
print('Harga tertinggi: ', harga.max())
print('Harga setelah diskon 10%: ', harga * 0.9)

Rata-rata harga:  6300.0
Harga tertinggi:  12000
Harga setelah diskon 10%:  [ 4500.  6300.  2700. 10800.  4050.]


## **Analisis 1**
Kalau menggunakan list biasa, [5000,7000,3000] * 0.9 akan error. Karena list Python tidak bisa langsung dikalikan dengan angka desimal.

#**Latihan 2**

In [ ]:
import pandas as pd

data_kantin = {
    'menu':['Nasi Goreng', 'Es Teh', 'Mie Ayam', 'Es Teh', None],
    'harga':[12000, 4000, 10000, 4000, 8000],
    'terjual':[23, 40, None, 35, 18]
}
df = pd.DataFrame(data_kantin)
print(df)

          menu  harga  terjual
0  Nasi Goreng  12000     23.0
1       Es Teh   4000     40.0
2     Mie Ayam  10000      NaN
3       Es Teh   4000     35.0
4         None   8000     18.0


## **Analisis 2**
Data kosong(none)
ada di kolom menu dan terjual. Kalau tidak dibersihkan, hasil analisis bisa kurang tepat dan bisa mengganggu perhitungan.

#**Latihan 3**

In [ ]:
df = pd.read_csv('/data_kantin - data_kantin.csv')

print(df.head())
print()
print(df.info())
print()
print(df.describe())
print()
print(df.shape)

      tanggal         menu kategori  harga  terjual
0  2026-08-20   Teh Hangat  Minuman   3000     11.0
1  2026-08-31  Nasi Goreng  Makanan  12000     29.0
2  2026-08-21  Nasi Goreng  Makanan  12000     28.0
3  2026-08-28   Teh Hangat  Minuman   3000     12.0
4  2026-08-21     Es Jeruk  Minuman   5000     19.0

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 110 entries, 0 to 109
Data columns (total 5 columns):
 #   Column    Non-Null Count  Dtype  
---  ------    --------------  -----  
 0   tanggal   110 non-null    object 
 1   menu      107 non-null    object 
 2   kategori  110 non-null    object 
 3   harga     110 non-null    int64  
 4   terjual   104 non-null    float64
dtypes: float64(1), int64(1), object(3)
memory usage: 4.4+ KB
None

              harga     terjual
count    110.000000  104.000000
mean    6336.363636   22.923077
std     3520.001232   10.831178
min     2000.000000    8.000000
25%     3000.000000   13.000000
50%     5000.000000   20.000000
75%    10000.00000

## **Analisis 3**
Kolom yang jumlah non-null-nya lebih sedikit dari jumlah baris adalah menu (107) dan terjual (104). Artinya, kedua kolom tersebut memiliki beberapa data yang kosong.

#**Latihan 4**

In [ ]:
print(df.isnull().sum())

df['terjual'] = df['terjual'].fillna(0)
df = df.dropna(subset=['menu'])

tanggal     0
menu        3
kategori    0
harga       0
terjual     6
dtype: int64


## **Analisis 4**
Kolom terjual diisi 0 supaya tetap bisa dihitung. Sedangkan menu yang kosong dihapus karena nama menu diperlukan untuk mengetahui data yang sedang dianalisis.

#**Latihan 5**

In [ ]:
print(df.duplicated().sum())
df = df.drop_duplicates()

df['harga'] = df['harga'].astype(int)
print(df.dtypes)

5
tanggal      object
menu         object
kategori     object
harga         int64
terjual     float64
dtype: object


/tmp/ipykernel_2794/3561001990.py:4: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df['harga'] = df['harga'].astype(int)


##**Analisis 5**
Jumlah baris berkurang jika terdapat data yang sama. Tipe data perlu diperiksa supaya perhitungan dan pengolahan data tidak mengalami masalah.

#**Latihan 6**

In [ ]:
laris = df[df['terjual'] > 20]
urut = df.sort_values(by='terjual', ascending=False)

df['total_pendapatan'] = df['harga'] * df['terjual']

ringkasan = df.groupby('menu')['total_pendapatan'].sum()
print(ringkasan)

menu
Es Jeruk       1460000.0
Es Teh         2144000.0
Kerupuk         972000.0
Mie Ayam       2410000.0
Nasi Goreng    3924000.0
Roti Bakar     1288000.0
Teh Hangat      459000.0
Name: total_pendapatan, dtype: float64


/tmp/ipykernel_2794/2047603725.py:4: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df['total_pendapatan'] = df['harga'] * df['terjual']


## **Analisis 6**
Menu yang menghasilkan total pendapatan tertinggi adalah Nasi Goreng, yaitu Rp3.924.000. Data ini bisa membantu kantin mengetahui menu yang menghasilkan pendapatan paling besar.

# **Tugas Kelompok**
**Anggota Kelompok**

1. Ni Putu Chalysta B.W.P/30

2.  Tatiana Beauty Octavia R./33


**Dataset yang dipilih:**
- Data Penjualan Toko Alat Tulis


**Pertanyaan Analisis Awal**
1. Produk apa yang paling banyak terjual?
2. Produk apa yang menghasilkan pendapat paling tinggi?
3. Apakah terdapat data kosong dan data duplikat?


**Dugaan Masalah Kualitas Data:**
- Missing value pada kolom Produk dan Terjual
- Kemungkinan terdapat data duplikat
- Tipe data Terjual masih berupa float



**Rencana Teknik Pembersihan:**
- Mengecek dan menangani data kosong, menghapus data duplikat, serta memastikan tipe data sudah sesuai.



**Rencana Manipulasi Data**
- Filter: Produk dengan jumlah terjual lebih dari 20
- Sort: Mengurutkan jumlah terjual dari terbesar ke terkecil
- Kolom turunan: Total pendapatan = Harga × Terjual
- Groupby/agregasi: Menghitung total pendapatan berdasarkan produk





**Jadwal kerja:**
- P3 (Loading & Inspection): Done
- P4 (Cleaning): Done
- P5  (Manipulation): Done
- P6 (Uji & Presentasi): ......


**Pembagian Peran**
- Anggota 1: Loading, inspection, dan cleaning data
- Anggota 2: Manipulation, analisis data, dan presentasi



In [ ]:
import pandas as pd

df = pd.read_csv('/EDA-KKA-DATASET.csv')

print("=== 5 DATA PERTAMA ===")
print(df.head())

print("\n=== INFO DATA ===")
print(df.info())

print("\n=== STATISTIK DATA ===")
print(df.describe())

print("\n=== UKURAN DATA ===")
print(df.shape)

# P4 - Cleaning
print("\n=== MISSING VALUE SEBELUM CLEANING ===")
print(df.isnull().sum())

df['Terjual'] = df['Terjual'].fillna(0)
df = df.dropna(subset=['Produk'])

print("\n=== MISSING VALUE SETELAH CLEANING ===")
print(df.isnull().sum())

print("\n=== DATA DUPLIKAT ===")
print("Jumlah duplikat:", df.duplicated().sum())

df = df.drop_duplicates()

print("Jumlah baris setelah cleaning:", len(df))

# P5 - Manipulation
print("\n=== FILTER: TERJUAL > 20 ===")
laris = df[df['Terjual'] > 20]
print(laris)

print("\n=== SORTING: TERJUAL TERBANYAK ===")
urut = df.sort_values(by='Terjual', ascending=False)
print(urut)

# Kolom turunan
df['total_pendapatan'] = df['Harga'] * df['Terjual']

print("\n=== DATA DENGAN TOTAL PENDAPATAN ===")
print(df)

# Groupby
print("\n=== TOTAL PENDAPATAN PER PRODUK ===")
ringkasan = df.groupby('Produk')['total_pendapatan'].sum()
print(ringkasan)

=== 5 DATA PERTAMA ===
      Tanggal       Produk       Kategori  Harga  Terjual
0  2026-09-03  Sticky Note  Perlengkapan    5000     19.0
1  2026-09-08       Spidol     Alat Tulis   5000     39.0
2  2026-09-03          Map   Perlengkapan   3000     37.0
3  2026-09-11   Buku Tulis           Buku   6000     38.0
4  2026-09-08   Kertas HVS         Kertas  45000     23.0

=== INFO DATA ===
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 50 entries, 0 to 49
Data columns (total 5 columns):
 #   Column    Non-Null Count  Dtype  
---  ------    --------------  -----  
 0   Tanggal   50 non-null     object 
 1   Produk    49 non-null     object 
 2   Kategori  50 non-null     object 
 3   Harga     50 non-null     int64  
 4   Terjual   49 non-null     float64
dtypes: float64(1), int64(1), object(3)
memory usage: 2.1+ KB
None

=== STATISTIK DATA ===
              Harga    Terjual
count     50.000000  49.000000
mean   10680.000000  24.163265
std    15194.910534   9.866498
min     2000.000000 